go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -62,25 +63,47 @@ func (c *addressManager) delAddr(ip net.IP) bool {
 }
 
 func (c *addressManager) Run(stopChan <-chan struct{}) {
-	errorCallback := func(err error) {
-		klog.Errorf("Failed during AddrSubscribe callback: %v", err)
-	}
+	var addrChan chan netlink.AddrUpdate
 	addrSubscribeOptions := netlink.AddrSubscribeOptions{
-		ErrorCallback: errorCallback,
+		ErrorCallback: func(err error) {
+			klog.Errorf("Failed during AddrSubscribe callback: %v", err)
+		},
 	}
-	addrChan := make(chan netlink.AddrUpdate)
-	if err := netlink.AddrSubscribeWithOptions(addrChan, stopChan, addrSubscribeOptions); err != nil {
-		klog.Errorf("Unable to run Node IP Manager, error during netlink subscribe: %v", err)
-		return
+
+	subScribeFcn := func() error {
+		addrChan = make(chan netlink.AddrUpdate)
+		if err := netlink.AddrSubscribeWithOptions(addrChan, stopChan, addrSubscribeOptions); err != nil {
+			return err
+		}
+		// sync the manager with current addresses on the node before we start processing events
+		c.sync()
+		return nil
 	}
 
-	// sync the manager with current addresses on the node before we start processing events
-	c.sync()
+	if err := subScribeFcn(); err != nil {
+		klog.Error("Unable to run Node IP Manager, error during netlink subscribe: %v", err)
+		return
+	}
 
 	go func() {
+		var gotInitialUpdate bool
+		initialSyncTimer := time.NewTimer(30 * time.Second)
+
 		for {
 			select {
-			case a := <-addrChan:
+			case a, ok := <-addrChan:
+				if !ok {
```

Comment:
this only covers when the channel is closed (which is the majority of what we care about), but additionally if we receive other types of errors we should just call c.sync() to verify we are not missing something because netlink coudln't handle a message properly. In the ErrorCallback I think you should call c.sync()

Reply:
TY @trozet! I pushed a revised version. Hopefully that covers it.